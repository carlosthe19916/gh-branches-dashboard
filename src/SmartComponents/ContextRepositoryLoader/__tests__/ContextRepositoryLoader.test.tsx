import React from "react";
import { shallow } from "enzyme";
import {
  ContextRepositoryLoader,
  ContextRepositoryLoaderStateToProps,
  ContextRepositoryLoaderDispatchToProps
} from "../ContextRepositoryLoader";
import { AxiosError } from "axios";

it("Test snapshot", () => {
  const stateToProps: ContextRepositoryLoaderStateToProps = {
    ctxRepositoryError: undefined,
    ctxRepositoryFetchStatus: "none"
  };
  const dispatchToProps: ContextRepositoryLoaderDispatchToProps = {
    fetchCtxRepository: jest.fn()
  };

  const wrapper = shallow(
    <ContextRepositoryLoader
      {...stateToProps}
      {...dispatchToProps}
      repositoryId={"carlosthe19916/gh-branches-dashboard"}
    >
      <p>my content</p>
    </ContextRepositoryLoader>
  );

  expect(wrapper).toMatchSnapshot();
});

it("Test snapshot :: fetch 'none'", () => {
  const stateToProps: ContextRepositoryLoaderStateToProps = {
    ctxRepositoryError: undefined,
    ctxRepositoryFetchStatus: "none"
  };
  const dispatchToProps: ContextRepositoryLoaderDispatchToProps = {
    fetchCtxRepository: jest.fn()
  };

  const wrapper = shallow(
    <ContextRepositoryLoader
      {...stateToProps}
      {...dispatchToProps}
      repositoryId={"carlosthe19916/gh-branches-dashboard"}
    >
      <p>my content</p>
    </ContextRepositoryLoader>
  );

  expect(wrapper).toMatchSnapshot();
});

it("Test snapshot :: fetch 'inProgress'", () => {
  const stateToProps: ContextRepositoryLoaderStateToProps = {
    ctxRepositoryError: undefined,
    ctxRepositoryFetchStatus: "inProgress"
  };
  const dispatchToProps: ContextRepositoryLoaderDispatchToProps = {
    fetchCtxRepository: jest.fn()
  };

  const wrapper = shallow(
    <ContextRepositoryLoader
      {...stateToProps}
      {...dispatchToProps}
      repositoryId={"carlosthe19916/gh-branches-dashboard"}
    >
      <p>my content</p>
    </ContextRepositoryLoader>
  );

  expect(wrapper).toMatchSnapshot();
});

it("Test snapshot :: fetch 'complete'", () => {
  const stateToProps: ContextRepositoryLoaderStateToProps = {
    ctxRepositoryError: undefined,
    ctxRepositoryFetchStatus: "complete"
  };
  const dispatchToProps: ContextRepositoryLoaderDispatchToProps = {
    fetchCtxRepository: jest.fn()
  };

  const wrapper = shallow(
    <ContextRepositoryLoader
      {...stateToProps}
      {...dispatchToProps}
      repositoryId={"carlosthe19916/gh-branches-dashboard"}
    >
      <p>my content</p>
    </ContextRepositoryLoader>
  );

  expect(wrapper).toMatchSnapshot();
});

it("Test snapshot :: error", () => {
  // Given
  const stateToProps: ContextRepositoryLoaderStateToProps = {
    ctxRepositoryError: {
      name: "Axios error",
      message: "Error fetching data",
      isAxiosError: true,
      config: {},
      toJSON: jest.fn()
    },
    ctxRepositoryFetchStatus: "none"
  };
  const dispatchToProps: ContextRepositoryLoaderDispatchToProps = {
    fetchCtxRepository: jest.fn()
  };

  // When
  const wrapper = shallow(
    <ContextRepositoryLoader
      {...stateToProps}
      {...dispatchToProps}
      repositoryId={"carlosthe19916/gh-branches-dashboard"}
    >
      <p>my content</p>
    </ContextRepositoryLoader>
  );

  // Then
  expect(wrapper).toMatchSnapshot();
});

it("Test snapshot :: error 'retry'", () => {
  // Given
  window.location.reload = jest.fn();  

  const stateToProps: ContextRepositoryLoaderStateToProps = {
    ctxRepositoryError: {
      name: "Axios error",
      message: "Error fetching data",
      isAxiosError: true,
      config: {},
      toJSON: jest.fn()
    },
    ctxRepositoryFetchStatus: "none"
  };
  const dispatchToProps: ContextRepositoryLoaderDispatchToProps = {
    fetchCtxRepository: jest.fn()
  };

  // When
  const wrapper = shallow(
    <ContextRepositoryLoader
      {...stateToProps}
      {...dispatchToProps}
      repositoryId={"carlosthe19916/gh-branches-dashboard"}
    >
      <p>my content</p>
    </ContextRepositoryLoader>
  );

  wrapper.find("#ContextRepositoryLoader_retryButton").simulate("click");

  // Then
  expect(window.location.reload).toHaveBeenCalled();
});
