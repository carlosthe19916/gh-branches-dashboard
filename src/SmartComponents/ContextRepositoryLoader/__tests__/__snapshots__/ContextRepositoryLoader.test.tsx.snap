// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot :: error 1`] = `
<PageSection>
  <Bullseye>
    <EmptyState
      variant="large"
    >
      <EmptyStateIcon
        icon={[Function]}
      />
      <Title
        headingLevel="h5"
        size="lg"
      >
        carlosthe19916/gh-branches-dashboard
        .
      </Title>
      <EmptyStateBody>
        Error fetching repository. Try to reload the page to fetch the data again. You can reload the page clicking on the button below.
      </EmptyStateBody>
      <Component
        id="ContextRepositoryLoader_retryButton"
        onClick={[Function]}
        variant="primary"
      >
        Retry
      </Component>
    </EmptyState>
  </Bullseye>
</PageSection>
`;

exports[`Test snapshot :: fetch 'complete' 1`] = `
<Fragment>
  <p>
    my content
  </p>
</Fragment>
`;

exports[`Test snapshot :: fetch 'inProgress' 1`] = `
<PageSection>
  <Bullseye>
    Fetching...
  </Bullseye>
</PageSection>
`;

exports[`Test snapshot :: fetch 'none' 1`] = `
<PageSection>
  <Bullseye>
    Waiting for fetch
  </Bullseye>
</PageSection>
`;

exports[`Test snapshot 1`] = `
<PageSection>
  <Bullseye>
    Waiting for fetch
  </Bullseye>
</PageSection>
`;
